rtic-macros/src/codegen/bindings/cortex.rs

```diff
@@ -29,11 +29,35 @@ fn is_exception(name: &Ident) -> bool {
             | "SysTick"
     )
 }
+
 pub fn interrupt_ident() -> Ident {
     let span = Span::call_site();
     Ident::new("interrupt", span)
 }
 
+pub fn check_stack_overflow_before_init(
+    _app: &App,
+    _analysis: &CodegenAnalysis,
+) -> Vec<TokenStream2> {
+    vec![quote!(
+        // Check for stack overflow using symbols from `cortex-m-rt`.
+        extern "C" {
+            pub static _stack_start: u32;
+            pub static __ebss: u32;
+        }
+
+        let stack_start = &_stack_start as *const _ as u32;
+        let ebss = &__ebss as *const _ as u32;
+
+        if stack_start > ebss {
+            // No flip-link usage, check the MSP for overflow.
+            if rtic::export::msp::read() <= ebss {
+                panic!("Stack overflow after allocating executors");
```

Comment:
Is there some kind of test suite that could be extended to verify that panic happens? I'm not sure is there's any "embedded-ci" like test suite that gates the master or not

Reply:
I tried a bit, but was not able to make a good test for this

Reply:
Np, maybe in the future then!